File: backend/src/main/java/com/lockin/lockin_app/entity/Badge.java
package com.lockin.lockin_app.entity;

import jakarta.persistence.*;

import lombok.Getter;
import lombok.Setter;

import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "badges")
@Getter
@Setter
public class Badge {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private String icon;

    @CreationTimestamp
    @Column(name = "earned_at", nullable = false, updatable = false)
    private LocalDateTime earnedAt;

    @Column(name = "badge_type", nullable = false)
    @Enumerated(EnumType.STRING)
    private BadgeType badgeType;
}

File: frontend/src/types/badge.ts
export interface Badge {
  id?: number;
  badgeType: string;
  name: string;
  description: string;
  icon: string;
  requirement: number;
  earnedAt?: string;
  earned: boolean;
}

export interface BadgeStats {
  totalBadges: number;
  earnedBadges: number;
  progressPercentage: number;
}
